src/TemplateManager.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace RebelCode\Bookings\Wordpress\Module;
+
+use Psr\Container\ContainerInterface;
+use Psr\EventManager\EventManagerInterface;
+
+/**
+ * Class TemplateManager
+ *
+ * Template manager is responsible for registering templates in given
+ * event manager from container.
+ *
+ * It register all defined templates as actions in event manager. All templates
+ * are prefixed with `eddbk` or any other given prefix. We need prefix when we
+ * tweaking from outside.
+ *
+ * In this module we can work without referencing to the
+ * prefix by using `render($templateActionName)` method.
+ *
+ * Inside render method we will apply prefix and trigger action in event manager after this.
+ *
+ * @package RebelCode\Bookings\Wordpress\Module
+ */
+class TemplateManager
+{
+    /**
+     * @var EventManagerInterface
+     */
+    protected $eventManager;
+
+    /**
+     * Template prefix
+     *
+     * @var
+     */
+    protected $prefix;
+
+    /**
+     * TemplateManager constructor.
+     *
+     * @param ContainerInterface $c
+     * @param string $prefix
+     * @throws \Psr\Container\ContainerExceptionInterface
+     * @throws \Psr\Container\NotFoundExceptionInterface
+     */
+    public function __construct(ContainerInterface $c, $prefix = 'eddbk')
+    {
+        $this->eventManager = $c->get('event-manager');
+        $this->prefix = $prefix;
+    }
+
+    /**
+     * Register all templates in event manager.
+     */
+    public function registerTemplates()
+    {
+        $this->_register(array_merge(
+            $this->_getGeneralTemplates(),
+            $this->_getServiceMetaboxTemplates(),
+            $this->_getBookingsTemplates()
+        ));
+    }
+
+    /**
+     * Render template by given template action.
+     *
+     * @param $template
+     * @return mixed
+     */
+    public function render($template)
+    {
+        return $this->eventManager->trigger(
+            $this->_makeTemplateActionName($template)
+        );
+    }
+
+    /**
+     * Register given templates in event manager.
+     *
+     * @param $templates
+     */
+    protected function _register($templates)
+    {
+        foreach ($templates as $template) {
+            $templateFilePath = __DIR__ . '/../templates/' . $template . '.phtml';

```

Comment:
I'd rather avoid using relative paths that traverse upwards in the file system. I recommend adding a constant in the `module.php` file that records the module's directory, such as

```
define('WP_BOOKINGS_UI_MODULE_DIR', __DIR__);
```

Then use this constant here.